test/test_mahalanobis_mixin.py

```diff
@@ -272,26 +272,25 @@ def test_get_squared_metric(estimator, build_dataset):
                          ids=ids_metric_learners)
 def test_transformer_is_2D(estimator, build_dataset):
   """Tests that the transformer of metric learners is 2D"""
-  # TODO: remove this check when SDML has become robust to 1D elements,
```

Comment:
in fact the pb was not robustness to 1D elements, but the fact that in 1D if the metric is [[negative number]], the algo will diverge (see above, since it's diagonal with negative coeff). This has been dealt with in the rest of the PR